src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1297,15 +1311,29 @@ public static TextLoader Create(IHostEnvironment env, ModelLoadContext ctx)
         }
 
         // These are legacy constructors needed for ComponentCatalog.
-        public static IDataLoader Create(IHostEnvironment env, ModelLoadContext ctx, IMultiStreamSource files)
+        internal static IDataLoader Create(IHostEnvironment env, ModelLoadContext ctx, IMultiStreamSource files)
             => (IDataLoader)Create(env, ctx).Read(files);
-        public static IDataLoader Create(IHostEnvironment env, Arguments args, IMultiStreamSource files)
+        internal static IDataLoader Create(IHostEnvironment env, Arguments args, IMultiStreamSource files)
             => (IDataLoader)new TextLoader(env, args, files).Read(files);
 
         /// <summary>
-        /// Convenience method to create a <see cref="TextLoader"/> and use it to read a specified file.
+        /// Creates a <see cref="TextLoader"/> and uses it to read a specified file.
+        /// </summary>
+        /// <param name="env">The environment to use.</param>
+        /// <param name="columns">Defines a mapping between input columns in the file and IDataView columns.</param>
+        /// <param name="hasHeader">Whether the file has a header.</param>
+        /// <param name="separatorChar"> The character used as separator between data points in a row. By default the tab character is used as separator.</param>
+        /// <param name="fileSource">Specifies a file from which to read.</param>
+        public static IDataView ReadFile(IHostEnvironment env, IMultiStreamSource fileSource, Column[] columns, bool hasHeader = false, char separatorChar = '\t')
```

Comment:
Why do we have both `IDataView ReadFromTextFile(this DataOperations catalog,` and these methods?  I think we should only have 1.